File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0;
        int bp = INT_MAX;
        
        for(int i = 0; i < prices.size(); i++){
            bp = min(bp, prices[i]);
            if(prices[i] > bp){
                profit = max(profit, prices[i]-bp);
            }
        }
        return profit;
    }
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        buyPrice = prices[0]
        profit = 0
        
        for i in range(1, len(prices)):
            if prices[i] < buyPrice:
                buyPrice = prices[i]
            else:
                profit = max(profit, prices[i]-buyPrice)

        return profit
    
    
    